protokube/pkg/protokube/aws_volume.go

```diff
@@ -164,73 +164,83 @@ func newEc2Filter(name string, value string) *ec2.Filter {
 func (a *AWSVolumes) findVolumes(request *ec2.DescribeVolumesInput) ([]*Volume, error) {
 	var volumes []*Volume
 	err := a.ec2.DescribeVolumesPages(request, func(p *ec2.DescribeVolumesOutput, lastPage bool) (shouldContinue bool) {
-		for _, v := range p.Volumes {
-			volumeID := aws.StringValue(v.VolumeId)
-			vol := &Volume{
-				ID: volumeID,
-				Info: VolumeInfo{
-					Description: volumeID,
-				},
-			}
-			state := aws.StringValue(v.State)
+		volumes = a.findEctdVolumes(p, volumes)
+		return true
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("error querying for EC2 volumes: %v", err)
+	}
+	return volumes, nil
+}
 
-			vol.Status = state
+func (a *AWSVolumes) findEctdVolumes(p *ec2.DescribeVolumesOutput, volumes []*Volume) []*Volume {
```

Comment:
Small typo on function name (`findEctdVolumes` -> `findEtcdVolumes`) and could include a comment for the function so go-lint doesn't warn on it.